File: app/src/main/java/adapters/WebsiteAdapter.java
package adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.sqube.tipshub.R;

import java.util.ArrayList;

import models.Website;

public class WebsiteAdapter extends RecyclerView.Adapter<WebsiteHolder> {
    ArrayList<Website> siteList;

    public WebsiteAdapter(ArrayList<Website> siteList){
        this.siteList = siteList;
    }

    @NonNull
    @Override
    public WebsiteHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.sport_site, parent, false);
        return new WebsiteHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull WebsiteHolder holder, int position) {
        holder.setDisplay(siteList.get(position));
    }

    @Override
    public int getItemCount() {
        if(siteList==null|| siteList.isEmpty())
            return 0;
        else
            return siteList.size();
    }
}

File: app/src/main/java/adapters/WebsiteAdapter.kt
package adapters

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.sqube.tipshub.R
import models.Website
import java.util.*

class WebsiteAdapter(var siteList: ArrayList<Website>?) : RecyclerView.Adapter<WebsiteHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): WebsiteHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_sport_site, parent, false)
        return WebsiteHolder(view)
    }

    override fun onBindViewHolder(holder: WebsiteHolder, position: Int) {
        holder.setDisplay(siteList!![position])
    }

    override fun getItemCount(): Int {
        return if (siteList == null || siteList!!.isEmpty()) 0 else siteList!!.size
    }
}